app/assets/javascripts/tags.js

```diff
@@ -73,6 +73,22 @@ $(() => {
     });
   });
 
+  function processSynonyms($search, synonyms) {
+    if (!synonyms) return synonyms;
+
+    if (synonyms.length > 3) {
+      const searchValue = $search.data('select2').selection.$search.val().toLowerCase();
+      displayedSynonyms = synonyms.filter(ts => ts.name.includes(searchValue)).slice(0, 3);
+    } else {
+      displayedSynonyms = synonyms;
```

Comment:
`displayedSynonyms` is missing a full declaration. You probably want to add `let displayedSynonyms;` before the if-else block, so that you can set it in each branch.